### ISSUE TITLE
salt-ssh problem possibly related to ipv6

### ISSUE THREAD

**Issue by tomasfejfar** [saltstack/salt#22984]
When I try to run salt-ssh on an ipv6 server I got:

```
$ salt-ssh myserver test.ping -l debug
myserver:
    deploy
... (with -l debug)
[DEBUG   ] STDOUT 2a02:2b88:2:1::3d3e:1
_edbc7885e4f9aac9b83b35999b68d015148caf467b78fa39c05f669c0ff89878
deploy

[DEBUG   ] STDERR 2a02:2b88:2:1::3d3e:1

[DEBUG   ] loading output in ['/var/cache/salt/master/extmods/output', '/usr/lib/python2.7/dist-packages/salt/output']
...
```

There is empty line after both STDOUT and STDERR. 

```
root@salt:~# salt-ssh --versions-report
           Salt: 0.17.5
         Python: 2.7.3 (default, Mar 13 2014, 11:03:55)
         Jinja2: 2.6
       M2Crypto: 0.21.1
 msgpack-python: 0.1.10
   msgpack-pure: Not Installed
       pycrypto: 2.6
         PyYAML: 3.10
          PyZMQ: 13.1.0
            ZMQ: 3.2.3
```

**Comment by tomasfejfar** [comment]
Target is `Linux vm15678 3.2.0-4-amd64 #1 SMP Debian 3.2.63-2+deb7u2 x86_64 GNU/Linux`

**Comment by jfindlay** [comment]
@tomasfejfar, 0.17.5 is a very old release and salt-ssh has been almost completely written since then.  Would you mind upgrading and see if you have the same problem?  Thanks.

**Comment by tomasfejfar** [comment]
Unfortunately not ATM. But it was fixed by adding a IPv4 address to the server.

**Comment by The-Loeki** [comment]
duplicate of #30561

**Comment by The-Loeki** [comment]
Can be closed now that #38831 is in

**Comment by rallytime** [comment]
Yes, I think so. Thanks for your work here @The-Loeki!
